### PR TITLE
Jiaruifang/exeception handler

### PR DESCRIPTION
**Pull request by feifeibear**
1. Deal with exception when chunk allocation failed.
2. rename always warmup as the static partition. In consist with the description in the paper.

### CODE REVIEW

File: examples/pretrain_bert_demo.py [hunk 1]
```diff
@@ -85,9 +85,9 @@ def _add_patrick_star_args(parser):
     )
     # Some hyperparams to tune when you failed to run a model.
     group.add_argument(
-        "--always_warmup",
+        "--with_static_partion",
```
Comment (zhuzilin) [inline examples/pretrain_bert_demo.py:88]: partion => partition

File: patrickstar/core/chunk_data.py [hunk 2]
```diff
@@ -113,31 +113,36 @@ def allocate_payload(self, device):
 
         NOTE() This method does not check availability. Please check if
         there is enough room for the chunk.
-
+        This function should be exeception-safe.
```
Comment (zhuzilin) [inline patrickstar/core/chunk_data.py:116]: exeception => exception

File: patrickstar/core/chunk_list.py [hunk 3]
```diff
@@ -166,6 +173,23 @@ def access_chunk(self, chunk_id: int, compute_device: torch.device):
         else:
             logger.debug(f"access_chunk chunk {chunk_id} already on {compute_device}")
 
+    def clear_useless_chunks(self, target_device: torch.device):
+        """
+        Move out all chunks not in comput on target_device.
```
Comment (zhuzilin) [inline patrickstar/core/chunk_list.py:178]: comput => compute

File: patrickstar/core/chunk_list.py [hunk 4]
```diff
@@ -154,7 +154,14 @@ def access_chunk(self, chunk_id: int, compute_device: torch.device):
...
             self.prepare_device(compute_device, payload_space)
-            chunk.allocate_payload(compute_device)
+            if chunk.allocate_payload(compute_device):
+                return
+            else:
+                self.clear_useless_chunks(compute_device)
+                if chunk.allocate_payload(compute_device) is False:
+                    raise RuntimeError(
+                        f"Allocate Payload Failed even if we have moved out more memory from {compute_device}"
+                    )
```
Comment (zhuzilin) [inline patrickstar/core/chunk_list.py:164]: 感觉可以把这个重试的这部分包成一个新的函数，因为下面也出现了类似的逻辑

File: patrickstar/core/memtracer/memtracer.py [hunk 5]
```diff
@@ -110,7 +110,7 @@ def __init__(self, local_rank: int = 0, config=None):
                 "warmup_gpu_chunk_mem_ratio", 0.1
             )
             self.use_fake_dist = config.get("use_fake_dist", False)
-            self.always_warmup = config.get("always_warmup", False)
+            self.with_static_partion = config.get("with_static_partion", False)
```
Comment (zhuzilin) [inline patrickstar/core/memtracer/memtracer.py:113]: partion => partition ?

**Review by zhuzilin — APPROVED** [review]
LGTM! few typos.

File: examples/pretrain_bert_demo.py (hunk 1 above)
Comment (feifeibear) [reply examples/pretrain_bert_demo.py:88]: DONE

File: patrickstar/core/chunk_data.py (hunk 2 above)
Comment (feifeibear) [reply patrickstar/core/chunk_data.py:116]: DONE

File: patrickstar/core/chunk_list.py (hunk 3 above)
Comment (feifeibear) [reply patrickstar/core/chunk_list.py:178]: DONE

File: patrickstar/core/chunk_list.py (hunk 4 above)
Comment (feifeibear) [reply patrickstar/core/chunk_list.py:164]: DONE

Comment (feifeibear) [reply patrickstar/core/chunk_list.py:164]: DONE

**Pull request merged**